nemo/collections/tts/modules/fastpitch.py

```diff
@@ -261,12 +276,37 @@ def forward(
             if self.learn_alignment and pitch.shape[-1] != pitch_predicted.shape[-1]:
```

Comment:
Whatever we do between the pitch and energy should be consistent. In which case I would remove the shape check here. Ideally the pitch and energy processing could just use the same helper method.

Reply:
Ideally we should, but I am sceptical removing a condition in the pitch workflow in an energy related PR (which makes it difficult to track changes), and also that would take some more testing for pitch. Can we take notes of the general refactoring and keep them for a different dedicated PR for refactoring?

Reply:
Fine by me